### PR TITLE
FlxSwipe: add released, path, qualified and velocity

### PR DESCRIPTION
**Pull request by DleanJeans**
For issue #1895
[Test case](https://gist.github.com/DleanJeans/f3c0261bc98e078b23eae92fcd1fef9a)
Tested on Flash and Android
It seems to have some problems with releasing touch/mouse (swipe never released)

### CODE REVIEW

**Comment by DleanJeans** [comment]
I think there won't be any problems with swiping now unless you tap crazily with many fingers on phone targets. Needs something to get rid of those never-get-released swipes.

Also CI has some problems with
```
The command "haxelib install task > /dev/null" failed and exited with 1 during .
```

**Comment by mastef** [comment]
Fighting atm with never-get-released swipes too, seems to happen when the finger leaves the game's bounds? ( e.g. left/right of screen ) - could this be?

File: flixel/input/FlxSwipe.hx [hunk 1]
```diff
@@ -3,60 +3,140 @@ package flixel.input;
...
+	
 	/**
 	 * Either LEFT_MOUSE, MIDDLE_MOUSE or RIGHT_MOUSE, 
 	 * or the touchPointID of a FlxTouch.
 	 */
 	public var ID(default, null):Int;
 	
-	public var startPosition(default, null):FlxPoint;
-	public var endPosition(default, null):FlxPoint;
+	public var path(default, null):FlxPath;
```
Comment (Gama11) [inline flixel/input/FlxSwipe.hx]: Creative use of `FlxPath`. Wouldn't a plain `Array<FlxPoint>` be more suitable though?

File: flixel/input/touch/FlxTouch.hx [hunk 2]
```diff
@@ -75,11 +77,19 @@ class FlxTouch extends FlxPointer implements IFlxDestroyable implements IFlxInpu
...
 			justPressedPosition.set(screenX, screenY);
 			justPressedTimeInTicks = FlxG.game.ticks;
+			#if FLX_POINTER_INPUT 
+			FlxG.swipes.push(_currentSwipe = new FlxSwipe(touchPointID, justPressedPosition, justPressedTimeInTicks)); 
+			#end
 		}
 		#if FLX_POINTER_INPUT
+		else if (pressed)
+		{
+			_currentSwipe.path.addPoint(getScreenPosition());
```
Comment (Gama11) [inline flixel/input/touch/FlxTouch.hx]: So the path will contain a point for each frame that the swipe was active for? I wonder if that isn't a bit much.

File: flixel/input/FlxSwipe.hx [hunk 3]
```diff
@@ -3,60 +3,140 @@ package flixel.input;
...
+	private function get_end():FlxPoint
+	{
+		return path.tail();
+	}
+	
+	private function get_velocity():FlxPoint
+	{
+		if (!released)
+		{
+			velocity.set(end.x - start.x, end.y - start.y);
```
Comment (Gama11) [inline flixel/input/FlxSwipe.hx]: I'm not sure if this definition of velocity where you just take the start and end position into account makes sense. You could have a very fast, but _circular_ swipe, meaning that `start` and `end` are virtually the same point.

**Comment by Gama11** [comment]
This implementation has a few issues as pointed out by my earlier comments.

I'd like to revisit this in the future though, there's definitely room for improvement in the `FlxSwipe` API, which is why I'm keeping #1895 open.

**Pull request closed without merging**